### DISCUSSION TITLE
Version check is failing on EMR cluster's spark version

### DISCUSSION THREAD

**Discussion by reygvasquez** [moj-analytical-services/splink#954] [Q&A] [answered]
When running splink on EMR cluster, the spark version value of '2.4.7-amzn-1' is considered invalid by version checker.  Any work around to this on the splink side?  If looks like the "-amzn-1" portion is what's causing failure, and I can't change what spark returns for version.

An error was encountered:
Invalid version: '2.4.7-amzn-1'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/splink/spark/spark_linker.py", line 154, in __init__
    if version.parse(self.spark.version) >= threshold and not self.catalog:
  File "/usr/local/lib/python3.7/site-packages/packaging/version.py", line 52, in parse
    return Version(version)
  File "/usr/local/lib/python3.7/site-packages/packaging/version.py", line 197, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
packaging.version.InvalidVersion: Invalid version: '2.4.7-amzn-1'

**Comment by RobinL** [comment, marked as answer]
Thanks for this, we will fix. If you downgrade to 3.4.5 for now it should work

**Reply by RobinL** [reply]
@reygvasquez this should now be fixed in [v3.5.2](https://github.com/moj-analytical-services/splink/releases/tag/v3.5.2)
